h2/src/main/org/h2/mvstore/type/DataType.java

```diff
@@ -22,51 +23,56 @@
      * @return -1 if the first key is smaller, 1 if larger, and 0 if equal
      * @throws UnsupportedOperationException if the type is not orderable
      */
-    int compare(Object a, Object b);
+    int compare(T a, T b);
```

Comment:
In needs `@Override`. The whole declaration can be removed too, but it that case the `UnsupportedOperationException` will be undocumented.